src/ntfsck.c

```diff
@@ -3144,14 +3183,39 @@ static void ntfsck_validate_index_blocks(ntfs_volume *vol,
 
 			for (; (u8 *)ie < index_end;
 				ie = (INDEX_ENTRY *)((u8 *)ie + le16_to_cpu(ie->length))) {
-				/* Bounds checks. */
-				if (((u8 *)ie < (u8 *)ia) ||
+
+				/* check bitmap for sub-node */
+				if (ie->ie_flags & INDEX_ENTRY_NODE) {
```

Comment:
I wonder that this change should be under /* check length bound */ ?

Reply:
moved it under /* check length bound */